### ISSUE TITLE
Elsewhen in an algorithm section does not work

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#9205]
### Description
Using `elsewhen` in an equation section works as expected, but not in an algorithm section. I have found nothing in the spec (e.g. 11.2.7.1 Where a When-Statement May Occur) that restricts use of elsewhen in an algorithm section.

### Steps to Reproduce
Run
```
model ElseWhenAlgorithm
//equation // this works, prints "time 0.1 time 0.5"
algorithm  // this breaks:
// Too few equations, under-determined system. The model has 1 equation(s) and 2 variable(s).
// Variable $whenCondition2 does not have any remaining equation to be solved in.
  when time > 0.1 then
    print("time 0.1 ");
  elsewhen time > 0.5 then
    print("time 0.5");
  end when;
end ElseWhenAlgorithm;
```

Using `equation` instead of algorithm works as expected.
Removing the `elsewhen` clause works as expected.
Splitting the when statement into two works as expected, but this way you can't implement when-equation priority:
``` 
  when time > 0.1 then
    print("time 0.1 ");
  end when;
  when time > 0.5 then
    print("time 0.5");
  end when;
```
### Expected Behavior
The model simulates and prints "time 0.1 time 0.5"

### Version and OS
OpenModelica 1.19.0, Windows 10

**Comment by casella** [comment]
I agree that elsewhen in algorithms is not forbidden by the specification. On the other hand, I understand it is really necessary in case of equations (that have no particular ordering, except the one induced by dependencies). If you write when statements in _algorithms_, the specification says that the algorithm is executed sequentially, so there is indeed some priority. This is a bit confusing in the specification, see modelica/ModelicaSpecification#3216.

**Comment by bilderbuchi** [comment]
> If you write when statements in algorithms, the specification says that the algorithm is executed sequentially, so there is indeed some priority.

Indeed, I missed that algorithm sections impose a natural ordering of two separate when statements. 

However, afaict the semantics of when-when and when-elsewhen are not the same (e.g. when both test expressions become true at the same time), so both have different applications.

Compare this arbitrary (untested) pseudo-code example
```
Boolean alarmstate = True;
Boolean heating = True;
Real heater_power;
algorithm
when alarmstate then
  print("Shutting down operation, something went wrong");
  heater_power = 0;
elsewhen heating then
  heater_power := 11;
end when;
``` 
and
```
Boolean alarmstate = True;
Boolean heating = True;
Real heater_power;
algorithm
when alarmstate then
  print("Shutting down operation, something went wrong");
  heater_power = 0;
end when;
when heating then
  heater_power := 11;
end when;
``` 
Running this code in my head, the former prints the message and sets power to 0, the latter prints the message, but the power ends up at 11.

**Comment by bilderbuchi** [comment]
> I'm investigating...

Ha! It seems the presence of only a print statement in both clauses triggers this bug, as soon as I start assigning variables this compiles.
```
model ElseWhenAlgorithm2
  Real foo;
algorithm
  when time > 0.1 then
    print("time 0.1 ");
    foo := 1;
  elsewhen time > 0.5 then
    print("time 0.5");
//    foo := 2; // or here
  end when;
end ElseWhenAlgorithm2;
```
However, only using asserts (what this is actually used like in XogenyTest) also fails to compile like the OP, so I'd still argue that this is a real bug:
```
model ElseWhenAlgorithm3
algorithm
  when time > 0.1 then
    assert(true, "trigger 0.1");
  elsewhen time > 0.5 then
    assert(false, "trigger 0.5");
  end when;
end ElseWhenAlgorithm3;
```

**Comment by casella** [comment]
@bilderbuchi also check Hans Olsson's proposal to fix the specification in modelica/ModelicaSpecification#3217
